src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -729,7 +729,7 @@ private static MethodDesc ResolveInterfaceMethodToVirtualMethodOnTypeRecursive(M
                     return null;
                 }
 
-                MethodDesc currentTypeInterfaceResolution = ResolveInterfaceMethodToVirtualMethodOnType(interfaceMethod, currentType);
+                MethodDesc currentTypeInterfaceResolution = ResolveInterfaceMethodToVirtualMethodOnType(interfaceMethod, currentType, returnRecursive: true);
                 if (currentTypeInterfaceResolution != null)
```

Comment:
There's still one branch that can return `null` in theory:
https://github.com/dotnet/runtime/blob/9b2032730355ca056a672bcb3afdff6444757e82/src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs#L659

I don't think it can happen in practice since it's likely impossible code path in the grand scheme of things but I didn't feel safe enough to reason about it.

Reply:
And if it does we go back through the loop with the base type… but didn’t we already do that in the first call? Is there a way this could return anything except null?

Reply:
Or, put another way, the `returnRecursive: true` indicates that **if** `x = ResolveInterfaceMethodToVirtualMethodOnTypeRecursive(interfaceMethod, baseType);` is computed **AND** `x is not null` then it changes the behavior to `return x;` instead of `return null`.

Previously it returned `null`, the loop proceeded to set `currentType = currentType.MetadataBaseType;` and continuing the loop would be equivalent to executing the recursive form `return ResolveInterfaceMethodToVirtualMethodOnTypeRecursive(interfaceMethod, currentType.MetadataBaseType);`.

The observation is that we just computed it and threw it away... so let's not throw it away.